src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx

```diff
@@ -60,63 +66,70 @@ const GeneralInfoStep = ({ btnsBox }) => {
-            label={'Last Name'}
-            onChange={() => handleInputChange(event, 'lastName')}
+          <AppTextField
+            errorMsg={t(errors.lastName)}
+            label={t('becomeTutor.generalInfo.inputLastName')}
+            onBlur={handleBlur('lastName')}
+            onChange={handleInputChange('lastName')}
             required
-            value={stepData?.generalInfo?.data?.lastName ?? ''}
+            type='text'
+            value={data.lastName}
           />
           <FormControl>
-            <InputLabel>Country</InputLabel>
-            <Select
-              label='Country'
-              onChange={(event) => handleInputChange(event, 'country')}
-              value={stepData?.generalInfo?.data?.country ?? ''}
-            >
-              {countries.map((c) => (
-                <MenuItem key={c.id} value={c.id}>
-                  {c.label}
-                </MenuItem>
-              ))}
-            </Select>
+            <AsyncAutocomplete
+              fetchOnFocus={!data.country}
+              onBlur={handleBlur('country')}
+              onChange={handleSelectChange('country')}
+              service={LocationService.getCountries}
+              textFieldProps={countriesOptions}
+              value={data.country}
+            />
           </FormControl>
-
           <FormControl>
-            <InputLabel>City</InputLabel>
-            <Select
-              label='City'
-              onChange={(event) => handleInputChange(event, 'city')}
-              value={stepData?.generalInfo?.data?.city ?? ''}
-            >
-              {cities.map((c) => (
-                <MenuItem key={c.id} value={c.id}>
-                  {c.label}
-                </MenuItem>
-              ))}
-            </Select>
+            <AsyncAutocomplete
+              fetchOnFocus={!data.city}
+              onBlur={handleBlur('city')}
+              onChange={handleSelectChange('city')}
+              service={getCities(data.country)}
+              textFieldProps={citiesOptions}
+              value={data.city}
+            />
           </FormControl>
         </Box>
-        <TextField
+        <AppTextField
+          errorMsg={t(errors.professionalSummary)}
           fullWidth
-          inputProps={{ maxLength: 70 }}
+          inputProps={{ maxLength: 101 }}
           multiline
-          onChange={() => handleInputChange(event, 'professionalSummary')}
+          onBlur={handleBlur('professionalSummary')}
+          onChange={handleInputChange('professionalSummary')}
           placeholder={t('becomeTutor.generalInfo.textFieldLabel')}
           rows={4}
-          value={stepData?.generalInfo?.data?.professionalSummary ?? ''}
+          type='text'
+          value={data.professionalSummary}
         />
         <Typography
           sx={styles.summaryLength}
-        >{`${stepData?.generalInfo?.data?.professionalSummary?.length}/70`}</Typography>
+        >{`${data.professionalSummary?.length}/100`}</Typography>
```

Comment:
I think `100` should be saved in some constant for using the same value (or +1) on 111 line.

Reply:
Thank you for the advice! I created a `maxLengthTextField` field in the `constants.js` file and imported this value into `GeneralInfoStep.jsx`.